### PR TITLE
Determine LXD socket before retrieving a client

### PR DESCRIPTION
**Pull request by MusicDin**
Instead of using LXD socket in snap's directory when an error contains `/var/lib/lxd/unix.socket` in the error, determine the LXD socket before retrieving the client.

### CODE REVIEW

**Comment by MusicDin** [comment]
@simondeziel @adamcstephens What do you think if we set `TerraformLxdConfig` for `local` during provider's initialization? This way there would be no need to set `LXD_DIR`/`LXD_SOCKET` environment variable before retrieving a client. However, I am not sure if this would have any negative implications on existing codebase..

File: lxd/provider.go [hunk 1]
```diff
@@ -766,3 +757,39 @@ func determineDaemonAddr(lxdRemote terraformLXDConfig) (string, error) {
...
+func determineLxdDir() (string, error) {
+	lxdSocket, ok := os.LookupEnv("LXD_SOCKET")
+	if ok {
+		return path.Dir(lxdSocket), nil
+	}
+
+	lxdDir, ok := os.LookupEnv("LXD_DIR")
+	if ok {
+		return lxdDir, nil
+	}
```
Comment (adamcstephens) [inline lxd/provider.go:773]: Should we also check this directory for a writable socket, or just assume the user knows what they're doing?

Comment (MusicDin) [reply lxd/provider.go:773]: Good point, we could check both LXD_SOCKET and LXD_DIR (if any of those is set) to fail faster. I'll fix that.

**Comment by simondeziel** [comment]
> @simondeziel @adamcstephens What do you think if we set `TerraformLxdConfig` for `local` during provider's initialization? This way there would be no need to set `LXD_DIR`/`LXD_SOCKET` environment variable before retrieving a client. However, I am not sure if this would have any negative implications on existing codebase..

I don't see any harm in exporting those 2 env variables.

**Review by simondeziel — APPROVED** [review]

**Comment by adamcstephens** [comment]
This isn’t Windows compatible so will need to be refactored. It’s blocking the release. https://github.com/terraform-lxd/terraform-provider-lxd/actions/runs/6698055575/job/18200325673

**Pull request merged**